File: Tool/IDL&Generator/DIDL/Test_proxy.cs
using System;
using System.Net;
using System.Collections.Generic;


using NGNet;
namespace TestC2S
{
	public class Proxy : NGNet.RmiProxy
	{
		public void LogInReq( RmiContext rmiContext, string deviceID, DateTime tmVaue, List<int> vecValue )
		{
			Message msg = new Message();
			UMessageMarshal.Write( msg, deviceID );
			UMessageMarshal.Write( msg, tmVaue );
			UMessageMarshal.Write( msg, vecValue );
			RmiSend( rmiContext, Common.Rmi_LogInReq, msg );
		}
		public void CheckServerInfoReq( RmiContext rmiContext )
		{
			Message msg = new Message();
			RmiSend( rmiContext, Common.Rmi_CheckServerInfoReq, msg );
		}
	}
}

namespace TestS2C
{
	public class Proxy : NGNet.RmiProxy
	{
		public void LogInAck( RmiContext rmiContext, string deviceID, DateTime tmValue, List<int> vecValue )
		{
			Message msg = new Message();
			UMessageMarshal.Write( msg, deviceID );
			UMessageMarshal.Write( msg, tmValue );
			UMessageMarshal.Write( msg, vecValue );
			RmiSend( rmiContext, Common.Rmi_LogInAck, msg );
		}
		public void CheckServerInfoAck( RmiContext rmiContext, int iError, EventInfo info )
		{
			Message msg = new Message();
			UMessageMarshal.Write( msg, iError );
			UMessageMarshal.Write( msg, info );
			RmiSend( rmiContext, Common.Rmi_CheckServerInfoAck, msg );
		}
	}
}

File: Tool/IDL&Generator/DIDL/Test_proxy.cpp
#pragma once

#include "Test_proxy.h"

namespace TestC2S
{
	void Proxy::LogInReq( HostID remote, NGNet::RmiContext rmiContext, const String &deviceID, const CNgTime &tmVaue, const std::vector<NGDungeonTime> &vecValue )
	{
		auto msg = CMessage::Create();
		msg->SetID( Rmi_LogInReq );
		*msg << deviceID;
		*msg << tmVaue;
		*msg << vecValue;
		RmiSend( remote, rmiContext, Rmi_LogInReq, msg );
	}
}

namespace TestS2C
{
	void Proxy::LogInAck( HostID remote, NGNet::RmiContext rmiContext, const String &deviceID, const CNgTime &tmValue, const std::vector<int> &vecValue )
	{
		auto msg = CMessage::Create();
		msg->SetID( Rmi_LogInAck );
		*msg << deviceID;
		*msg << tmValue;
		*msg << vecValue;
		RmiSend( remote, rmiContext, Rmi_LogInAck, msg );
	}
}

